cinema/views.py

```diff
@@ -1 +1,77 @@
-# write views here
+from rest_framework import viewsets
+
+from cinema.models import (
+    Movie,
+    MovieSession,
+    Genre,
+    CinemaHall,
+    Actor
+)
+from cinema.serializers import (
+    MovieSerializer,
+    MovieListSerializer,
+    MovieDetailSerializer,
+    MovieSessionSerializer,
+    MovieSessionListSerializer,
+    MovieSessionDetailSerializer,
+    GenreSerializer,
+    CinemaHallSerializer,
+    ActorSerializer,
+)
+
+
+class GenreViewSet(viewsets.ModelViewSet):
+    queryset = Genre.objects.all()
+    serializer_class = GenreSerializer
+
+
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.all()
+
+    def get_queryset(self):
+        queryset = self.queryset
+
+        if self.action in ("list", "retrieve"):
+            queryset = queryset.prefetch_related("actors", "genres")
+
+        return queryset
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return MovieListSerializer
+
+        if self.action == "retrieve":
+            return MovieDetailSerializer
+
+        return MovieSerializer
+
+
+class MovieSessionViewSet(viewsets.ModelViewSet):
+    queryset = MovieSession.objects.all()
+
+    def get_queryset(self):
+        queryset = self.queryset
+
+        if self.action in ("movie", "cinema_hall"):
+            queryset = queryset.select_related("movie", "cinema_hall")
+
+        return queryset
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return MovieSessionListSerializer
+
+        if self.action == "retrieve":
+            return MovieSessionDetailSerializer
```

Comment:
Create a dict for this, otherwise, you always need to add 2 new lines with a new action:
```
serializers = {
    "key_is_action": value_is_serializer
}
```
And then: `serializers.get(self.action, DefaultSerializer)`